api/server/server.go

```diff
@@ -43,15 +43,22 @@ type Server struct {
 	Datastore    models.Datastore
 }
 
-func New(ctx context.Context, ds models.Datastore, mq models.MessageQueue, r *runner.Runner, tasks chan task.Request, enqueue models.Enqueue) *Server {
+type Components struct {
```

Comment:
Like I said the idea of this PR is just make it simple and clean.

We need to remember that the idea of IronFunctions is planned to be an customizable/extensible server. So if we have an API that embeds many other components (current situation, can probably get worse), the user creating its own server will have provide each other component to the API Server in its instantiation.

I agree we need to make `server` package smaller. But still the `server` will be the main component on this server and it must have access to other replaceable components.

I agree solution this adds the problem of having to assert if all components has been correctly provided in the `server.Component` (easy thing to solve IMO)

But what other solution do you have for these problems and also respect the idea of a `IronFunctions API` *extensible and customizable*?